### PR TITLE
Added CommandClient, Update README with latest examples, remove client from Adapters, simplify Intents

### PR DESCRIPTION
**Pull request by DjDeveloperr**
In this PR, I have
- [Commands] Added Command Client! Built-in Command Framework to ease making Commands, without having to write a Parser from scratch.
  - Supports a variety of options
  - Completely Object-oriented
  - Easy to use
- [Client] Added a simple way to use Intents. A class with three static members - All, Presence, GuildMembers, which make it easier to pass intents with client.
- [Caching] Removed client from Cache Adapters as it was unnecessary.
- [Examples] Added run-anywhere example in /examples, which can be directly run from URL without any environment setup.
- [Gateway] When receiving HELLO, client now immediately sends Heartbeat for ping to be identified as soon as possible.
- [README] Updated README with features, updated examples and usage, and a small note regarding it being in development.
- [Self-bot] Partial Support for Self-bots. Their Gateway events don't work at the moment.

### CODE REVIEW

File: src/test/index.ts [hunk 1]
```diff
@@ -1,15 +1,16 @@
-import { Client } from '../models/client.ts'
-import { GatewayIntents } from '../types/gateway.ts'
+// import { Client } from '../models/client.ts'
```
Comment (AkiaCode) [inline src/test/index.ts]: Please delete unused imports

**Review by AkiaCode — CHANGES_REQUESTED** [review]

File: src/test/index.ts (hunk 1 above)
Comment (everlasting393) [reply src/test/index.ts]: 😂

File: README.md [hunk 2]
```diff
@@ -1,10 +1,16 @@
 # discord-deno
 
-![banner](images/discord-deno.png)
```
Comment (AkiaCode) [inline README.md:3]: Will be added later

File: examples/ping.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { Client, Message, Intents } from '../mod.ts'
+
+const client = new Client();
+
+client.on("ready", () => {
+    console.log(`Logged in as ${client.user?.tag}!`);
+});
+
+client.on("messageCreate", (msg: Message) => {
```
Comment (AkiaCode) [inline examples/ping.ts:9]: I hope you can comment on it so that others can understand

File: src/test/index.ts [hunk 4]
```diff
@@ -1,39 +1,25 @@
...
-// import { Embed } from "../structures/embed.ts"
-// import { Guild } from "../structures/guild.ts"
-import { Client, GatewayIntents, Message, DefaultCacheAdapter, ClientPresence, Member, Role, GuildChannel, TextChannel, Embed, Guild } from '../../mod.ts';
 
-const bot = new Client({
+const client = new Client({
   presence: new ClientPresence({
     activity: {
-      name: "Testing",
+      name: "Pokémon Sword",
```
Comment (AkiaCode) [inline src/test/index.ts]: What?

Comment (everlasting393) [reply src/test/index.ts]: Probably when he was testing the bot activity things

Comment (DjDeveloperr) [reply src/test/index.ts]: 👀 I thought its alright

File: examples/ping.ts (hunk 3 above)
Comment (DjDeveloperr) [reply examples/ping.ts:9]: Isn't it pretty straightforward? Still, I can do.

Comment (AkiaCode) [reply examples/ping.ts:9]: I think it will be okay if there is some

File: README.md (hunk 2 above)
Comment (AkiaCode) [reply README.md:3]: Path: banner.png

Comment (DjDeveloperr) [reply README.md:3]: I'll add it with next commit

Comment (AkiaCode) [reply README.md:3]: Thanks

File: examples/ping.ts [hunk 5]
```diff
@@ -21,8 +21,8 @@ if(!token) {
     Deno.exit();
 }
 
-const intents = prompt("Input Intents (0 = All, 1 = Presence, 2 = Server Members):");
-if(!intents || !["0", "1", "2"].includes(intents)) {
+const intents = prompt("Input Intents (0 = All, 1 = Presence, 2 = Server Members, 3 = None):");
+if(!intents || !["0", "1", "2", "3"].includes(intents)) {
```
Comment (AkiaCode) [inline examples/ping.ts:25]: Change `if(` to `if (`

File: examples/ping.ts [hunk 6]
```diff
@@ -32,8 +32,10 @@ if(intents == "0") {
     ints = Intents.All;
 } else if(intents == "1") {
     ints = Intents.Presence;
-} else {
+} else if(intents == "2") {
```
Comment (AkiaCode) [inline examples/ping.ts:35]: Change `if(` to `if (`

File: src/gateway/index.ts [hunk 7]
```diff
@@ -227,7 +230,24 @@ class Gateway {
         ),
         presence: this.client.presence.create()
       }
-    })
+    }
+
+    if(this.client.bot === false) {
```
Comment (AkiaCode) [inline src/gateway/index.ts:235]: Change `if(` to `if (`

File: src/models/rest.ts [hunk 8]
```diff
@@ -175,11 +176,29 @@ export class RESTManager {
...
     }
 
-    return {
+    let data: { [name: string]: any } = {
       headers,
       body: body?.file ?? JSON.stringify(body),
       method: method.toUpperCase()
     }
+
+    if(this.client.bot === false) {
```
Comment (AkiaCode) [inline src/models/rest.ts:185]: Change `if(` to `if (`

File: src/models/rest.ts [hunk 9]
```diff
@@ -230,9 +249,14 @@ export class RESTManager {
                   )
                   .join('&')
               : ''
-          const urlToUse =
+          let urlToUse =
             method === 'get' && query !== '' ? `${url}?${query}` : url
 
+          if(this.client.canary) {
```
Comment (AkiaCode) [inline src/models/rest.ts:255]: Change `if(` to `if (`

**Review by AkiaCode — CHANGES_REQUESTED** [review]

**Comment by DjDeveloperr** [comment]
I have fixed Lint-errors, and all checks have passed, I guess it's okay now.

**Review by Helloyunho — APPROVED** [review]
Good

**Pull request merged**